grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1132,106 +1140,438 @@ class StockMovementService {
+        StockMovement stockMovement = command.stockMovement
 
-                if (!availableItem) {
-                    String lotNumberName = data.lotNumber ?: g.message(code: "default.noLotNumber.label", default: Constants.DEFAULT_LOT_NUMBER)
-                    String binLocationName = data.binLocation ?: g.message(code: "default.noBinLocation.label", default: Constants.DEFAULT_BIN_LOCATION_NAME)
-                    data.errors.rejectValue(
-                            "id",
-                            "importPickCommand.availableItem.notFound.error",
-                            [lotNumberName, binLocationName] as Object[],
-                            "There is no item available with: lot \"${lotNumberName}\" and bin location \"${binLocationName}\""
-                    )
-                }
+        // The parent requisition items that picklist items are associated with
+        List<PickPageItem> pickPageItems = command.pickPageItems
+
+        // FIXME Remove this if it's no longer being used
+        // Configuration property that allows overpick for a given location
+        Boolean supportsOverpick = command.location.supports(ActivityCode.ALLOW_OVERPICK)
 
-                if (availableItem && AvailableItemStatus.listUnavailable().contains(availableItem.status)) {
-                    String lotNumberName = data.lotNumber ?: g.message(code: "default.noLotNumber.label", default: Constants.DEFAULT_LOT_NUMBER)
-                    String binLocationName = data.binLocation ?: g.message(code: "default.noBinLocation.label", default: Constants.DEFAULT_BIN_LOCATION_NAME)
+        // TODO Do we handle the case where ID is null?
+        // Group the picklist items from the CSV by requisition item ID
+        Map<String, List> picklistItemsGroupedByRequisitionItem = command.picklistItems.groupBy { it.id }
 
+        // Iterate over the parent requisition items
+        picklistItemsGroupedByRequisitionItem.each { requisitionItemId, picklistItems ->
+
+            // Iterate over each separate pick
+            picklistItems.each { PicklistItemCommand data ->
+
+                // FIXME Make sure we also check that the quantity provided is not greater than the
+                //  quantity requested. Should add this to the ImportPickCommand validator.
+                Integer quantityRequired = data.quantity
+
+                // FIXME Add proper comment to explain what this is for
+                // Base validation, creates errors object?
+                data.validate()
+
+                // TODO What happens if ID is null. We should throw an exception if we don't have a good answer.
+                //  skip validation if id is empty since most of the validation relies on an existing requisition id
+                // skip validation if id is empty since mos tof the validation relies on an existing requisition id
+                if (!data.id) {
```

Comment:
looks like it cannot be empty because id on `PicklistItemCommand` has constraints: `nullable: false, blank: false`